src/OpenLoco/src/GameCommands/Track/CreateTrackMod.cpp

```diff
@@ -0,0 +1,96 @@
+#include "CreateTrackMod.h"
+#include "Economy/Expenditures.h"
+#include "Map/TileManager.h"
+#include "Map/Track/TrackData.h"
+#include "Map/TrackElement.h"
+#include "Ui/WindowManager.h"
+#include "Vehicles/Vehicle.h"
+#include <OpenLoco/Math/Vector.hpp>
+
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::GameCommands
+{
+    // 0x004A6479
+    static currency32_t createTrackMod(const TrackModsPlacementArgs& args, uint8_t flags)
+    {
+        setExpenditureType(ExpenditureType::Construction);
+        setPosition(args.pos + World::Pos3(16, 16, 0));
+
+        auto* elTrack = [&args]() -> const World::TrackElement* {
```

Comment:
what's the point of making this a lambda instead of a normal function? it makes it harder to read the code here IMO